### PR TITLE
Handle zero bytes available in channel_read_map

### PR DESCRIPTION
**Pull request by aliddell**
- Fixes a bug in `reader_initialize()`.
- Update cycle and position for the channel hold, but do not set the channel state to `Mapped`, if no bytes are available to read.

### CODE REVIEW

File: src/runtime/channel.h [hunk 1]
```diff
@@ -17,14 +17,34 @@ extern "C"
...
+        size_t head;
+
+        /// Highest position in the channel that has been written to in the current cycle.
+        size_t high;
+
+        /// Number of times the buffer has been filled and wrapped around to the start.
+        size_t cycle;
+
+        /// Pointer to end position of reserved region before committing a write.
+        size_t mapped;
```
Comment (aliddell) [inline src/runtime/channel.h:37]: @nclack wording here is awkward, but the best I can think of. If you've got a better way to put it, feel free to commit it.

Comment (nclack) [reply src/runtime/channel.h:37]: Positions on the queue are described by both an offset in the buffer and the cycle count. I think of `head` and `cycle` as part of a tuple defining a particular position.

So, those fields might be best logically grouped.

But I think these descriptions are fine!

File: src/runtime/channel.h [hunk 2]
```diff
@@ -17,14 +17,34 @@ extern "C"
...
+        /// Position in the channel where the next write operation will occur.
+        size_t head;
+
+        /// Highest position in the channel that has been written to in the current cycle.
+        size_t high;
+
+        /// Number of times the buffer has been filled and wrapped around to the start.
+        size_t cycle;
+
+        /// Pointer to end position of reserved region before committing a write.
```
Comment (nclack) [inline src/runtime/channel.h:36]:
```suggestion
        /// Pointer to the end position of the reserved region of a mapped write.
```

**Review by nclack — APPROVED** [review]
lgtm! Left some suggestions that you can take or leave.

**Pull request merged**